src/js/osweb/items/loop.js

```diff
@@ -79,15 +78,14 @@ export default class Loop extends Item {
 
   /** Reset all item variables to their default value. */
   reset () {
-    this.matrix = []
+    this.orig_matrix = []

```

Comment:
It is really hard to follow what you are doing here, or why the orig_matrix is required to make resumptions possible, but I assume you tested it and this works. If so, I am ok with it.

Reply:
The `orig_matrix` is the original state of the matrix before any of the operations are applied. And then the `matrix` is the matrix as it used during a particular run of the `loop` item. This allows the loop item to run multiple times from the same state.